region/norfair/east/Bubble Mountain.json

```diff
@@ -4152,11 +4210,184 @@
       "link": [9, 9],
       "name": "Waver Farm",
       "requires": [
-        {"resetRoom": {"nodes": [2, 3]}},
-        {"partialRefill": {"type": "Energy", "limit": 100}},
-        {"partialRefill": {"type": "Missile", "limit": 6}}
+        {"or": [
+          {"and": [
+            {"resetRoom": {"nodes": [2]}},
+            {"cycleFrames": 660}
+          ]},
+          {"and": [
+            {"resetRoom": {"nodes": [3]}},
+            {"cycleFrames": 720}
+          ]}
+        ]}
+      ],
+      "resetsObstacles": ["A"],
+      "farmCycleDrops": [{"enemy": "Waver", "count": 2}]
+    },
+    {
+      "link": [9, 9],
+      "name": "Sova Farm",
+      "requires": [
+        {"or": [
+          {"and": [
+            {"resetRoom": {"nodes": [1]}},
+            {"or": [
+              {"and": [
+                "SpaceJump",
+                {"cycleFrames": 1500}
+              ]},
+              {"and": [
+                "HiJump",
+                "canWalljump",
```

Comment:
Should probably be at least `canPreciseWalljump`, maybe `canTrickyWalljump` because of the repetition?